### PR TITLE
GCW-3569 load offer and item messages seperately

### PR DESCRIPTION
**Pull request by swatijadhav**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3569

### What does this PR do?
Load messages on offers and items separately.

### CODE REVIEW

File: app/routes/review_offer.js [hunk 1]
```diff
@@ -39,6 +39,18 @@ export default AuthorizeRoute.extend({
...
+  afterModel(model) {
+    this.store.query("message", {
+      messageable_type: "Offer",
+      messageable_id: model.get("id")
+    });
+
+    this.store.query("message", {
+      messageable_type: "Item",
+      messageable_id: model.get("items").getEach("id")
+    });
```
Comment (patrixr) [inline app/routes/review_offer.js:51]: same as in the donor app, we likely want to await those promises

**Review by patrixr — CHANGES_REQUESTED** [review]

**Review by steveyken — APPROVED** [review]

File: app/routes/review_offer.js [hunk 2]
```diff
@@ -22,11 +23,22 @@ export default AuthorizeRoute.extend({
...
+
+    return new AjaxPromise(
+      `/offers/${offerId}`,
+      "GET",
+      _this.get("session.authToken"),
+      { exclude_messages: "true" }
+    ).then(function(data) {
+      _this.store.pushPayload(data);
+      return _this.store.peekRecord("offer", offerId);
+    });
```
Comment (patrixr) [inline app/routes/review_offer.js:41]:
```suggestion
    return new AjaxPromise(
      `/offers/${offerId}`,
      "GET",
      _this.get("session.authToken"),
      { exclude_messages: "true" }
    ).then((data) => {
      this.store.pushPayload(data);
      return this.store.peekRecord("offer", offerId);
    });
```

`let _this = this;` is a very old pre-lambda pattern, which we will never need to ever use again :)

Comment (patrixr) [reply app/routes/review_offer.js:41]: also, you can probably use `await` instead of `.then`

Comment (patrixr) [reply app/routes/review_offer.js:41]: (none of this is a big deal, just side notes for the future)

**Review by patrixr — APPROVED** [review]

**Pull request merged**